File: game-server/src/main/java/net/dodian/uber/game/model/player/skills/prayer/Prayer.java
package net.dodian.uber.game.model.player.skills.prayer;

import net.dodian.uber.game.model.entity.player.Client;
import net.dodian.uber.game.model.player.packets.outgoing.SendMessage;
import net.dodian.uber.game.model.player.skills.Skills;

/**
 * @author Dashboard
 */
public class Prayer {

    public static boolean buryBones(Client client, int itemId, int itemSlot) {
        Bones bone = Bones.getBone(itemId);
        if (bone == null || !client.playerHasItem(itemId))
            return false;
        client.requestAnim(827, 0);
        client.giveExperience(bone.getExperience(), Skills.PRAYER);
        client.deleteItem(itemId, itemSlot, 1);
        client.send(new SendMessage("You bury the " + client.GetItemName(itemId).toLowerCase()));
        return true;
    }

    public static boolean altarBones(Client client, int itemId) {
        Bones bone = Bones.getBone(itemId);
        if (bone == null || !client.playerHasItem(itemId)) {
            client.boneItem = -1;
            return false;
        }
        client.deleteItem(itemId, 1);
        client.requestAnim(3705, 0);
        double extra = (double) (client.getLevel(Skills.PRAYER) + 1) / 100;
        double chance = 2.0 + extra;
        client.giveExperience((int) (bone.getExperience() * chance), Skills.PRAYER);
        client.send(new SendMessage("The gods are pleased with your offering."));
        client.triggerRandom((int) (bone.getExperience() * chance));
        return true;
    }

    public static boolean chaosAltarBones(Client client, int itemId) {
        Bones bone = Bones.getBone(itemId);
        if (bone == null || !client.playerHasItem(itemId)) {
            client.boneItem = -1;
            return false;
        }
        client.deleteItem(itemId, 1);
        client.requestAnim(3705, 0);
        double extra = (double) (client.getLevel(Skills.PRAYER) + 1) / 100;
        double chance = 3.5 + extra;
        client.giveExperience((int) (bone.getExperience() * chance), Skills.PRAYER);
        client.send(new SendMessage("Zamorak smiles down upon you and blesses your offering."));
        client.triggerRandom((int) (bone.getExperience() * chance));
        return true;
    }
}

File: game-server/src/main/java/net/dodian/uber/game/model/player/skills/prayer/Prayer.kt
package net.dodian.uber.game.model.player.skills.prayer

import net.dodian.uber.game.model.entity.player.Client
import net.dodian.uber.game.model.player.skills.Skills
import java.util.*
import net.dodian.uber.api.Animation

fun buryBones(client: Client, itemId: Int, itemSlot: Int): Boolean {
        val bone = Bones.getBone(itemId)
        if (bone == null || !client.playerHasItem(itemId)) return false
        client.requestAnim(Animation.BURY_BONE, 0)
        client.giveExperience(bone.experience, Skills.PRAYER)
        client.deleteItem(itemId, itemSlot, 1)
        client.sendMessage("You bury the " + client.GetItemName(itemId).lowercase(Locale.getDefault()))
        return true
    }

    fun altarBones(client: Client, itemId: Int): Boolean {
        client.sendMessage("The gods are pleased with your offering.")
        doAction(client, itemId, 2.0)
        return true
    }

    fun chaosAltarBones(client: Client, itemId: Int): Boolean {
        client.sendMessage("Zamorak smiles down upon you and blesses your offering.")
        doAction(client, itemId, 3.5)
        return true
    }

    fun doAction(client: Client, itemId: Int, i: Double): Boolean {
        val bone = Bones.getBone(itemId)
        if (bone == null || !client.playerHasItem(itemId)) {
            client.boneChaos = -1
            client.boneAltar = -1
            return false
        }
        client.deleteItem(itemId, 1)
        client.requestAnim(Animation.OFFER_BONES_TO_ALTER, 0)
        val extra = (client.getLevel(Skills.FIREMAKING) + 1).toDouble() / 100
        val chance = i + extra
        client.giveExperience(((bone.experience * chance).toInt()), Skills.PRAYER)
        client.triggerRandom((Bones.getBone(itemId).experience * chance).toInt())
        return true
    }